cumulusci/cli/cci.py

```diff
@@ -1345,30 +1343,55 @@ def task_list(runtime, plain, print_json):
 
 
 @task.command(name="doc", help="Exports RST format documentation for all tasks")
-@pass_runtime(require_project=False)
-def task_doc(runtime):
-    config_src = runtime.universal_config
-
-    click.echo("==========================================")
-    click.echo("Tasks Reference")
-    click.echo("==========================================")
-    click.echo("")
-
-    for name, options in config_src.tasks.items():
+@click.option("--project", "project", is_flag=True, help="Write project specific tasks")
+@click.option(
+    "--write",
+    "write",
+    is_flag=True,
+    help="If project flag is toggled will write to ./docs/project_tasks.rst otherwise will write by default to ./docs/cumulusci_tasks.rst",
+)
+@pass_runtime(
+    require_project=False,
+)
+def task_doc(runtime, project=False, write=False):
+    result = ["=========================================="]
+    config = runtime.universal_config
+    tasks = config.tasks.items()
+    file_name = "./docs/cumulusci_tasks.rst"
+    # setting to read for project specific configuration
+    if project:
+        config = runtime.project_config.config
+        project_name = runtime.project_config.project__name.capitalize()
+        tasks = config["tasks"].items()
+        file_name = "./docs/project_tasks.rst"
+    # handling for general and project specific documentation
+    result.append(f"{project_name} Tasks Reference") if project else result.append(
+        "Tasks Reference"
+    )
+    result += [
+        "==========================================",
+        "",
+    ]  # extra space for formatting
+    for name, options in tasks:
         task_config = TaskConfig(options)
         doc = doc_task(name, task_config)
-        click.echo(doc)
-        click.echo("")
+        result += [f"{doc}", ""]
+    result = "\r\n".join(result)
+    if write:
+        Path("docs").mkdir(exist_ok=True)
+        with open(file_name, "w", encoding="utf-8") as f:
```

Comment:
We could reduce clutter at the top by moving the `file_name` var down here closer to where it's used. It's then also easier to reference in the code.
```suggestion
        Path("docs").mkdir(exist_ok=True)
        file_name = "project_tasks" if project else "cumulusci_tasks"
        with open(f"./docs/{file_name}.rst", "w", encoding="utf-8") as f:
```